lib/src/protocol/http/parser/mod.rs

```diff
@@ -599,43 +648,43 @@ pub enum HeaderResult<T> {
     Error,
 }
 
-impl<'a> Header<'a> {
+impl Header {
     pub fn value(&self) -> HeaderValue {
-        if compare_no_case(self.name, b"host") {
+        if compare_no_case(&self.name, b"host") {
             //FIXME: UTF8 conversion should be unchecked here, since we already checked the tokens?
-            if let Ok(s) = str::from_utf8(self.value).map(String::from) {
+            if let Ok(s) = str::from_utf8(&self.value).map(String::from) {
                 HeaderValue::Host(s)
             } else {
                 HeaderValue::Error
             }
-        } else if compare_no_case(self.name, b"content-length") {
-            if let Ok(l) = str::from_utf8(self.value) {
+        } else if compare_no_case(&self.name, b"content-length") {
+            if let Ok(l) = str::from_utf8(&self.value) {
                 if let Ok(length) = l.parse() {
                     return HeaderValue::ContentLength(length);
                 }
             }
             HeaderValue::Error
-        } else if compare_no_case(self.name, b"transfer-encoding") {
-            if compare_no_case(self.value, b"chunked") {
+        } else if compare_no_case(&self.name, b"transfer-encoding") {
+            if compare_no_case(&self.value, b"chunked") {
```

Comment:
this is where the value should be trimmed:
First, by making an iterator over the comma separated values, as is done for the [Connection header](https://github.com/sozu-proxy/sozu/pull/739/files#diff-aaf4cdf4ffb037e558c0d168e251c445f6249e0d2b78590ec9ca96d3a8942cbfR681) (I don't know why I did not do that long ago, and why we did not receive any bug report for that over the years...), and second by trimming left and right the values produced before comparing them to `chunked` and others